aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsEndpointDecorator.kt

```diff
@@ -127,6 +130,25 @@ class AwsEndpointDecorator : ClientCodegenDecorator {
         }
     }
 
+    override fun extraSections(codegenContext: ClientCodegenContext): List<Pair<DetachedSection<*>, (Section) -> Writable>> {
+        return regionalized(codegenContext).thenSingletonListOf {
+            SdkConfigSection.create { section ->
+                {
+                    rust(
+                        """
+                        ${section.serviceConfigBuilder}.set_aws_endpoint_resolver(${section.sdkConfig}.endpoint_resolver().clone());
```

Comment:
Is setting the `endpoint_resolver` here going to trigger a deprecation warning?

Reply:
this doesn't change the generated code—just changes where it's being generated from